rust/src/mask/masking.rs

```diff
@@ -75,16 +77,17 @@ impl Into<MaskObject> for Aggregation {
 #[allow(clippy::len_without_is_empty)]
 impl Aggregation {
     /// Creates a new, empty aggregator for masks or masked models.
-    pub fn new(config: MaskConfig) -> Self {
+    pub fn new(config: MaskConfig, object_size: usize) -> Self {
         Self {
             nb_models: 0,
             object: MaskObject::new(config, vec![]),
+            object_size,
```

Comment:
just a thought: could an alternative be to allocate the vec already here, e.g. `Vec::with_capacity(object_size)`? then we wouldn't need the `object_size` field anymore, I think (at the cost of an earlier allocation).

Reply:
What we can do is:
```
object: MaskObject::new(config, Vec::with_capacity(object_size)]),
object_size,
```
but we can't remove `object_size`because capacity and length are two different things. 
Capacity is the amount of space allocated for any future elements and length is the number of actual elements within the vector. [docs.rs](https://doc.rust-lang.org/std/vec/struct.Vec.html#capacity-and-reallocation)

Reply:
you're right, but after we set
```rust
object: MaskObject::new(config, Vec::with_capacity(object_size)),
```
later on, we can check against `self.object.data.capacity()`. we don't ever have more than `object_size` elements in `object`, so its capacity should also stay fixed. perhaps there are some downsides I haven't thought of though...

Reply:
Yeah since we never shrink or grow the vec, it _should_ be fine. But in theory, the capacity can change behind the scene so I'd rather not rely on that. That could lead to very hard to debug.